src/cephadm/cephadm

```diff
@@ -5452,6 +5548,593 @@ def command_gather_facts():
+
+    @property
+    def port_active(self):
+        return port_in_use(self.port)
+
+    @property
+    def can_run(self):
+        # if port is in use
+        if self.port_active:
+            self.errors.append(f"TCP port {self.port} already in use, unable to bind")
+        if not os.path.exists(os.path.join(self.daemon_path, CephadmDaemon.key_name)):
+            self.errors.append(f"Key file '{CephadmDaemon.key_name}' is missing from {self.daemon_path}")
+        if not os.path.exists(os.path.join(self.daemon_path, CephadmDaemon.crt_name)):
+            self.errors.append(f"Certificate file '{CephadmDaemon.crt_name}' is missing from {self.daemon_path}")
+        if self.token == "Unknown":
+            self.errors.append(f"Authentication token '{CephadmDaemon.token_name}' is missing from {self.daemon_path}")
+        return len(self.errors) == 0
+
+    @staticmethod
+    def _unit_name(fsid, daemon_id):
+        return "{}.service".format(get_unit_name(fsid, CephadmDaemon.daemon_type, daemon_id))
+
+    @property
+    def unit_name(self):
+        return CephadmDaemon._unit_name(self.fsid, self.daemon_id)
+
+    @property
+    def daemon_path(self):
+        return os.path.join(
+            args.data_dir,
+            self.fsid,
+            f'{self.daemon_type}.{self.daemon_id}'
+        )
+
+    @property
+    def binary_path(self):
+        return os.path.join(
+            args.data_dir,
+            self.fsid,
+            CephadmDaemon.bin_name
+        )
+
+    def _scrape_host_facts(self, refresh_interval=10):
+        ctr = 0
+        while True:
+            
+            if self.stop:
+                break
+
+            if ctr >= refresh_interval:
+                ctr = 0
+                logger.debug("executing host-facts scrape")
+                errors = []
+                s_time = time.time()
+                facts = HostFacts()
+                elapsed = time.time() - s_time
+                try:
+                    data = json.loads(facts.dump())
+                except json.decoder.JSONDecodeError:
+                    errors.append("host-facts provided invalid JSON")
+                    logger.warning(errors[-1])
+                    data = {}
+                self.cephadm_cache.update_task(
+                    'host',
+                    {
+                        "scrape_timestamp": s_time,
+                        "scrape_duration_secs": elapsed,
+                        "scrape_errors": errors,
+                        "data": data,                    
+                    }
+                )
+                logger.debug(f"completed host-facts scrape - {elapsed}s")
+            
+            time.sleep(CephadmDaemon.loop_delay)
+            ctr += CephadmDaemon.loop_delay
+        logger.info("host-facts thread stopped")
+
+    def _scrape_ceph_volume(self, refresh_interval=15):
+        args.command = "inventory --format=json".split()
+        args.fsid = self.fsid
+        args.log_output = False
```

Comment:
`args` is used several times in this PR but I have a hard time figuring out what this does. Here we assign three values and then never do anything with it. This suggests to me that some not so optimal abstraction is used...some magic global variable maybe?

Reply:
args is set for the invocation of the ceph-volume command, which uses the args to determine the mode of execution.expects the args to be defined. This way the entrypoint for invoking ceph-volume remains consistent with a normal cephadm command